common/fixation_entry.go

```diff
@@ -268,10 +494,18 @@ func (fs *FixationStore) getUnmarshaledEntryForBlock(ctx sdk.Context, safeIndex
 		fs.cdc.MustUnmarshal(iterator.Value(), &entry)
 
 		if entry.Block <= block {
-			if entry.IsStale(ctx) {
+			// stale entries work as markers of the extent to where the preceding
+			// (non stale) entry is valid. So if we meet one, we bail.
+			// however if that stale entry is also deleted (which means, it is the
+			// latest too) then we need to not bail, and do return that entry, to
+			// comply with the expecations of AppendEntry(). thus, we may return an
+			// entry that is both stale and deleted.
+			// for this reason, we explicitly test for a deleted entry in EntryGet()
+			// and AppendEntry(), and for stale entry in EntryFind(). so EntryGet()
+			// and EntryFind() would return not-found, and AppendEntry would fail.
+			if entry.IsStaleBy(blockHeight) && !entry.IsDeletedBy(blockHeight) {
```

Comment:
does this mean we can't use FindEntry to fetch a stale entry?
we want to be able to find stale entries, that is the point of keeping them in the state